app/frontend/src/api/api.ts

```diff
@@ -32,23 +37,58 @@ export async function askApi(request: ChatAppRequest, idToken: string | undefine
     return parsedResponse as ChatAppResponse;
 }
 
-export async function configApi(idToken: string | undefined): Promise<Config> {
-    const response = await fetch(`${BACKEND_URI}/config`, {
-        method: "GET",
-        headers: getHeaders(idToken)
-    });
-
-    return (await response.json()) as Config;
-}
-
 export async function chatApi(request: ChatAppRequest, idToken: string | undefined): Promise<Response> {
     return await fetch(`${BACKEND_URI}/chat`, {
         method: "POST",
-        headers: getHeaders(idToken),
+        headers: { ...getHeaders(idToken), "Content-Type": "application/json" },
         body: JSON.stringify(request)
     });
 }
 
 export function getCitationFilePath(citation: string): string {
     return `${BACKEND_URI}/content/${citation}`;
 }
+
+export async function uploadFileApi(request: FormData, idToken: string): Promise<SimpleAPIResponse> {
+    const response = await fetch("/upload", {
+        method: "POST",
+        headers: getHeaders(idToken),
```

Comment:
needed to add "Content-Type: application/json" here?

Reply:
No, this is actually why I needed to refactor, as upload does not send JSON. The response is JSON but the content-type is about the request.

Reply:
Resolving this, please re-open if there's still a question here.